Firestore/core/src/firebase/firestore/util/filesystem_posix.cc

```diff
@@ -130,51 +130,82 @@ Status DeleteFile(const Path& path) {
 }
 
 Status RecursivelyDeleteDir(const Path& parent) {
```

Comment:
This now seems totally portable, so it can move into `filesystem_common.cc`.

Reply:
`DeleteDir` is not portable though, right? Wouldn't I need to add that to the common header?

Reply:
`DeleteDir` is non-portable, but that's OK. There's three parts to making this work:
  * The public interface in `filessystem.h`
  * Extra low-level parts with a portable interface in `filesystem_detail.h`
  * `filesystem_common.cc` implements portably in terms of elements detail, e.g. [`RecursivelyCreateDir`](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/util/filesystem_common.cc#L27).

My suggestion here is that now that there's a portable iterator, this can work like `RecursivelyCreateDir` does.